crates/core/tedge/src/cli/connect/command.rs

```diff
@@ -274,7 +274,7 @@ fn check_device_status_c8y(tedge_config: &TEdgeConfig) -> Result<DeviceStatus, C
             Ok(Event::Outgoing(Outgoing::PingReq)) => {
                 // No messages have been received for a while
                 println!("Local MQTT publish has timed out.");
-                break;
+                return Err(ConnectError::TimeoutElapsedError);
```

Comment:
This fix is unappropriated. 
The problem is Line 293 returns `Ok(DeviceStatus::Unknown)`. Then, from Line 76 in the function `execute()`, we consider all `Ok(_)` case means successful.

```
                return match self.check_connection(&config) {
                    Ok(_) => {
                        let cloud = br_config.cloud_name.clone();
                        println!("Connection check to {} cloud is successfull.\n", cloud);
                        Ok(())
                    }
                    Err(err) => Err(err.into()),
                };
```

Think what we do in `tedge connect c8y` now. We first create a device in c8y, then we always do connection check. I think `DeviceStatus::Unknown` is not ok. It's an error.

Reply:
I agree, I fixed the logic.

Reply:
I would like to print the `connection check failed` message when the status returned is `Ok(DeviceStatus::Unknown)`. 
Retaining this state because this DeviceStatus will be returned when the local mqtt broker acknowledged (Puback) for the JWT request. But, we did not receive the response for the jwt mqtt request.